config/transmorpher.php

```diff
@@ -83,17 +83,38 @@
 
     /*
     |--------------------------------------------------------------------------
-    | Video Codec
+    | Video Decoder
     |--------------------------------------------------------------------------
     |
-    | The codec used when transcoding videos to HLS and DASH.
-    | This does not affect the codec used for MP4, which is x264 since it's the only one supported by the PHP-FFmpeg package for MP4.
+    | The decoder used when transcoding videos.
     |
     | You can choose from:
-    | x264, hevc
+    | cpu, nvidia_cuda
+    */
+    'decoder' => [
+        'name' => env('TRANSMORPHER_VIDEO_DECODER', 'cpu'),
+        'nvidia' => [
+            'hwFrames' => env('TRANSMORPHER_VIDEO_DECODER_NVIDIA_HWFRAMES', 10),
+        ],
+    ],
+
+    /*
+    |--------------------------------------------------------------------------
+    | Video Encoder
+    |--------------------------------------------------------------------------
     |
+    | The encoder used when transcoding videos.
+    |
+    | You can choose from:
+    | cpu_h264, cpu_hevc, nvidia_h264, nvidia_hevc
     */
-    'video_codec' => 'x264',
+    'encoder' => [
+        'name' => env('TRANSMORPHER_VIDEO_ENCODER', 'cpu_h264'),
+        'bitrate' => env('TRANSMORPHER_VIDEO_ENCODER_BITRATE', '6000k'),
```

Comment:
move settings into encoder specific files, with precedence over our implementation, but not over the init/additional from this file